blocks/library/embed/index.js

```diff
@@ -21,57 +23,125 @@ registerBlock( 'core/embed', {
 	category: 'common',
 
 	attributes: {
-		url: attr( 'iframe', 'src' ),
 		title: attr( 'iframe', 'title' ),
 		caption: children( 'figcaption' ),
 	},
 
-	edit( { attributes, setAttributes, focus, setFocus } ) {
-		const { url, title, caption } = attributes;
+	edit: class extends wp.element.Component {
+		constructor() {
+			super( ...arguments );
+			this.doServerSideRender = this.doServerSideRender.bind( this );
+			// Copies the block's url so we can edit it without having the block update
+			this.state = {
+				oembed_html: '',
+				error: false,
+				fetching: false,
+			};
+			this.noPreview = [ 'facebook.com', 'www.facebook.com' ];
+		}
 
-		if ( ! url ) {
-			return (
-				<Placeholder icon="cloud" label={ wp.i18n.__( 'Embed URL' ) } className="blocks-embed">
-					<input type="url" className="placeholder__input" placeholder={ wp.i18n.__( 'Enter URL to embed here...' ) } />
-					<Button isLarge>
-						{ wp.i18n.__( 'Embed' ) }
-					</Button>
-				</Placeholder>
+		doServerSideRender( event ) {
```

Comment:
Could we extract this ServerSideRender to a HoC? Maybe something like:

```js
<ServerSideBlock blockType={ blockType } arguments={ arguments }>
  { ( html, error ) => (
    <Sandbox html={ html } /> 
  ) }
</ServerSideBlock>
```

Any blocker to achieve this?